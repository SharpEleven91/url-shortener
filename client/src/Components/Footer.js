import React from "react";
import { Code } from "@material-ui/icons";
import { IconButton } from "@material-ui/core";
const Footer = () => {
  return (
    <IconButton
      data-testid="github-button"
      href="https://github.com/SharpEleven91/url-shortener"
      id="github-link"
    >
      <Code fontSize="large" />
    </IconButton>
  );
};

export default Footer;
