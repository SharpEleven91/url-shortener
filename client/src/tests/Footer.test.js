import React from 'react';
import expect from 'expect';
import Footer from '../Components/Footer';
import { render, cleanup } from 'react-testing-library';
import "jest-dom/extend-expect";
afterEach(cleanup);
test('Component renders properly', () => {
    const { queryByTestId } = render(<Footer/>);
    expect(queryByTestId("github-button")).toBeInTheDocument();
    expect(queryByTestId("github-button")).toMatchSnapshot();
});
