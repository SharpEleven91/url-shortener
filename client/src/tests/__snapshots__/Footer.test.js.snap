// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly 1`] = `
<a
  class="MuiButtonBase-root-7 MuiIconButton-root-1"
  data-testid="github-button"
  href="https://github.com/SharpEleven91/url-shortener"
  id="github-link"
  role="button"
  tabindex="0"
>
  <span
    class="MuiIconButton-label-6"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-10 MuiSvgIcon-fontSizeLarge-18"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
      />
    </svg>
  </span>
  <span
    class="MuiTouchRipple-root-19"
  />
</a>
`;
